package/src/utils/add-virtual-import.ts

```diff
@@ -0,0 +1,93 @@
+import { type Plugin } from 'vite'
+import type { HookParameters } from "astro";
+
+function resolveVirtualModuleId<T extends string>(id: T): `\0${T}` {
+	return `\0${id}`
+}
+
+function moduleBoilerplate(content: object): string {
+    return `
+        export default ${ JSON.stringify(content) }
+    `
+}
+
+function createVirtualModule(name: string, content: string | object): Plugin {
+    const pluginName = `vite-plugin-${ name }`
+    const virtualModuleName = `virtual:${ name }`
```

Comment:
I'd prefer the name without any default prefix. There are valid usecases for non-prefixed virtual imports, eg. virtual imports meant to be used by users